File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch05/Python/P05-Exp.py
# Evaluates the exponential function
from math import *
from elemfunc import *

# main

for i in range(-20,25,5):
  x = float(i)
  expx = exp(x)                                                    # built-in
  Expx = Exp(x)                                                # power series
  Exp1x = Exp1(x)                                        # continued fraction
  print(x,Exp1x,1e0-Exp1x/expx)

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch05/C/P05-Exp.cpp
// Evaluates the exponential function
#include <stdio.h>
#include <math.h>
#include "elemfunc.h"

int main()
{
   double expx, Expx, Exp1x, x;
   
   for (x=-5e0; x<=5e0; x+=1e0) {
      expx = exp(x);                                              // built-in
      Expx = Exp(x);                                          // power series
      Exp1x = Exp1(x);                                  // continued fraction
      printf("%10f %15.7e %15.7e %15.7e %15.7e\n",
             x,expx,Expx,Exp1x,1e0-Exp1x/expx);
   }
}
